### PR TITLE
Kubernetes events generation for domain resource changes

### PR DESCRIPTION
**Pull request by sankarpn**
The PR adds the integration tests to check the domain events logged by the operator for changes in domain custom resource.
The test verifies the events for DomainCreated, DomainChanged,  DomainDeleted, DomainProcessingStarting, DomainProcessingCompleted, DomainProcessingRetrying, and DomainProcessingAborted events.

https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/3433/

### CODE REVIEW

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/utils/K8sEvents.java [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+          if (event.getReason().contains(reason)
+              && event.getMetadata().getCreationTimestamp().isAfter(timestamp.getMillis())) {
+            logger.info(Yaml.dump(event));
+            verifyOperatorDetails(event, opNamespace, domainUid);
+            //verify reason
+            logger.info("Verifying domain event {0}", reason);
+            assertTrue(event.getReason().equals(reason));
+            //verify messages
+            logger.info("Verifying domain event message {0}", getDomainEventMessage(reason, domainUid));
+            assertTrue(event.getMessage().contains(getDomainEventMessage(reason, domainUid)));
```
Comment (rjeberhard) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/utils/K8sEvents.java:59]: I'm concerned about having the integration test look for a specific message text. This seems too tightly coupled. What do others think? Would it be enough to have the test look for a reason code?

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItKubernetesEvents.java [hunk 2]
```diff
@@ -0,0 +1,385 @@
...
+  }
+
+  /**
+   * Test domain events are logged when domain resource goes through various life cycle stages.
+   * Verifies DomainCreated is logged when domain resource is created.
+   * Verifies DomainProcessingStarting is logged when operator processes the domain resource.
+   * Verifies DomainProcessingCompleted is logged when operator done processes the domain resource.
+   * Verifies DomainChanged is logged when operator processes the domain resource changes.
+   * Verifies DomainProcessingRetrying is logged when operator retries the failed domain resource changes.
+   * Verifies DomainProcessingAborted is logged when operator exceeds the maximum retries.
```
Comment (doxiao) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItKubernetesEvents.java:136]: We need to test DomainProcessingFailed event as well.

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/utils/K8sEvents.java [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+        break;
+      case DOMAIN_PROCESSING_STARTING:
+        message = "Creating or updating Kubernetes presence for WebLogic Domain with UID " + domainUid;
+        break;
+      case DOMAIN_PROCESSING_COMPLETED:
+        message = "Successfully completed processing domain resource " + domainUid;
+        break;
+      case DOMAIN_PROCESSING_FAILED:
+        message = "Failed to process domain resource " + domainUid
+            + " due to xxxx, the processing  will be retried if required";
```
Comment (doxiao) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/utils/K8sEvents.java:122]: The xxxx in the message needs to be replace with the real error message.

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/utils/K8sEvents.java (hunk 1 above)
Comment (doxiao) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/utils/K8sEvents.java:59]: I am okay with just checking the reason code; the unit tests check the messages (and other fields) already.

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItKubernetesEvents.java (hunk 2 above)
Comment (doxiao) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItKubernetesEvents.java:136]: In addition, owls-72816 added another event - DomainValidationError. We need integration tests to cover that event as well, but it does not have to be part of this PR.

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/utils/K8sEvents.java [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+  }
+
+  /**
+   * Returns the description for different event types.
+   *
+   * @param reason event to check for Created, Changed, deleted, processing etc
+   * @param domainUid UID of the domain
+   * @return String containing the complete description
+   */
+  public static String getDomainEventMessage(String reason, String domainUid) {
```
Comment (doxiao) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/utils/K8sEvents.java:99]: now that we don't check event messages anymore, do we still need this method?

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItKubernetesEvents.java (hunk 2 above)
Comment (sankarpn) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItKubernetesEvents.java:136]: I will add it part of another PR

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/utils/K8sEvents.java (hunk 3 above)
Comment (sankarpn) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/utils/K8sEvents.java]: removed

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/utils/K8sEvents.java (hunk 4 above)
Comment (sankarpn) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/utils/K8sEvents.java]: removed

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/utils/K8sEvents.java (hunk 1 above)
Comment (sankarpn) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/utils/K8sEvents.java]: removed the message checking

**Review by doxiao — APPROVED** [review]

**Review by rjeberhard — APPROVED** [review]

**Pull request merged**